n3fit/src/n3fit/model_gen.py

```diff
@@ -76,8 +80,16 @@ def _generate_experimental_layer(self, pdf):
         else:
             output_layers = [obs(pdf) for obs in self.observables]
 
-        # Finally concatenate all observables (so that experiments are one single entitiy)
-        ret = op.concatenate(output_layers, axis=2)
+        masked_output_layers = []
+
+        for output_layer, mask_layer in zip(output_layers, self.trvl_mask_layers):
+            if mask_layer is None:
+                masked_output_layers.append(output_layer)
+            else:
+                masked_output_layers.append(mask_layer(output_layer))
+
+        # Finally concatenate all observables (so that experiments are one single entity)
+        ret = op.concatenate(masked_output_layers)
```

Comment:
Why is the axis removed? (I guess the default is exactly the right axis, but I'd like to have it explicit, it makes debugging easier)

Reply:
I thought it was a bit explicit, but since all tensor shapes have to be as explicit as possible for tensorflow to do the correct thing, I will re-insert it.

Reply:
More than for tensorflow is for the person reading the code in this case, since it is hard to keep track of which axis is what :P